runtime/src/dpe_crypto.rs

```diff
@@ -78,64 +101,171 @@ impl<'a> Crypto for DpeCrypto<'a> {
 
     fn derive_cdi(
```

Comment:
I think we may want to implement the `drop` trait for DpeCrypto and use it to clear the keyvault slots we used when the struct goes out of scope. drop is similar to a destructor.

```rust
impl Drop for DpeCrypto {
    fn drop(&mut self) {...}
}
```